### PR TITLE
Don't do duplicate tx_type rdo with bottom-up partition rdo

### PR DESCRIPTION
**Pull request by ycho**
Fixing #598 and #580.

### CODE REVIEW

**Comment by ycho** [comment]
[AWCY results says no chance in bdrate](https://beta.arewecompressedyet.com/?job=master-91ab6d2d1215c9acf449bcbbc92a88842edc83f9&job=test_deduplicate_tx_type_rdo_bottom%402018-10-09T22%3A27%3A39.917Z)

|   PSNR | PSNR Cb | PSNR Cr | PSNR HVS |   SSIM | MS SSIM | CIEDE 2000 |
|   ---: |    ---: |    ---: |     ---: |   ---: |    ---: |       ---: |
| 0.0000 |  0.0000 |  0.0000 |   0.0000 | 0.0000 |  0.0000 |     0.0000 |

**Comment by ycho** [comment]
There is 0.39% regression with -s 0, according to [AWCY results](https://beta.arewecompressedyet.com/?job=master-91ab6d2d1_s0_5f%402018-10-09T23%3A43%3A01.847Z&job=test_deduplicate_tx_type_rdo_bottom_s0_5f%402018-10-09T23%3A43%3A22.174Z)

|   PSNR | PSNR Cb | PSNR Cr | PSNR HVS |   SSIM | MS SSIM | CIEDE 2000 |
|   ---: |    ---: |    ---: |     ---: |   ---: |    ---: |       ---: |
| 0.3964 |  0.1491 |  0.2035 |   0.1458 | 0.7616 |  0.6051 |     0.2466 |

**Comment by rzumer** [comment]
This was attempted before with the same results. I think that @fbossen can explain why it is necessary to perform the transform RDO again. Like I suggested before, a comment in the code would be helpful to justify it so that someone does not attempt to remove it again (I think it is at least the 3rd time that someone makes that attempt).

**Comment by ycho** [comment]
I agree as Frank think that top-down partition rdo requires to re-compute tx_type.
However, I am trying to figure out why it is required with bottom-up partition.
If the reason has been explained before, I'd appreciate if you remind me.

**Comment by lu-zero** [comment]
The tests fail...

**Comment by ycho** [comment]
results of "cargo test --release --features=decode_test -- --ignored" 

running 5 tests
test test_encode_decode::high_bd ... FAILED
test test_encode_decode::quantizer ... ok
test test_encode_decode::odd_size_frame_with_full_rdo ... FAILED
test test_encode_decode::dimensions ... ok
test test_encode_decode::speed ... FAILED

**Comment by ycho** [comment]
Yay! Figured out how to remove the duplicated rdo_tx_size_type() call for bottom-up partition!

Two reasons:
1) If skip is true, rdo mode decision should use tx_type == DCT_DCT
2) when  rdo_tx_size_type() is removed in bottom-up partition rdo, should call set_tx_size(.., tx_size),
since this has been only called in rdo_tx_size_type(), which are used by deblocking.

For speed=0 and 5 frames, no bdrate change, as shown by[ AWCY results](https://beta.arewecompressedyet.com/?job=master-c6d1669_s0_5f%402018-10-10T21%3A44%3A45.817Z&job=test3_deduplicate_tx_type_rdo_bottom_s0_5f%402018-10-10T22%3A11%3A07.269Z)

|   PSNR | PSNR Cb | PSNR Cr | PSNR HVS |   SSIM | MS SSIM | CIEDE 2000 |
|   ---: |    ---: |    ---: |     ---: |   ---: |    ---: |       ---: |
| 0.0000 |  0.0000 |  0.0000 |   0.0000 | 0.0000 |  0.0000 |     0.0000 |

**Comment by tmatth** [comment]
> ~5% faster !?

encoding-time

**Comment by ycho** [comment]
@tmatth, thanks Tristian for importing the speed graph!

File: src/rdo.rs [hunk 1]
```diff
@@ -354,6 +356,8 @@ pub fn rdo_mode_decision(
         let wr: &mut dyn Writer = &mut WriterCounter::new();
         let tell = wr.tell_frac();
 
+        if skip { tx_type = TxType::DCT_DCT; };
```
Comment (xiphmont) [inline src/rdo.rs]: Is this a bug fix?

Comment (xiphmont) [reply src/rdo.rs]: heh, oops, midair collision.  Question still stands :-)

Comment (ycho) [reply src/rdo.rs]: It IS a FIX! :) Since the tx_type, by def of av1, should be DCT_DCT if skip is true.

**Pull request merged**